### ISSUE TITLE
`src and type required `

### ISSUE THREAD

**Issue by ehmicky** [Tom-Bonnike/netlify-plugin-inline-critical-css#8]
The following error is happening in some builds:

```
Failed to inline critical CSS.
`src` and `type` required 
    /opt/buildhome/.netlify-build-plugins/node_modules/oust/index.js:52:15 oust
    /opt/buildhome/.netlify-build-plugins/node_modules/oust/index.js:82:37 Function.module.exports.raw
    /opt/buildhome/.netlify-build-plugins/node_modules/critical/src/file.js:354:28 getStylesheetHrefs
    /opt/buildhome/.netlify-build-plugins/node_modules/critical/src/file.js:839:32 getDocument
    /opt/buildhome/.netlify-build-plugins/node_modules/critical/src/core.js:108:26 async create
    /opt/buildhome/.netlify-build-plugins/node_modules/critical/index.js:23:20 async Object.generate
    /opt/buildhome/.netlify-build-plugins/node_modules/netlify-plugin-inline-critical-css/index.js:28:9 async onPostBuild
    src/plugins/child/run.js:18:3 async Object.run
    src/plugins/child/main.js:38:38 async handleEvent
    src/plugins/ipc.js:97:9 async process.<anonymous>
```

The repository is private. The build logs do not contain any additional information unfortunately. Do you know what the problem might be?

**Comment by Tom-Bonnike** [comment]
After digging a bit in the `critical` & `oust` source code, it seems like the problem is that the `src` passed to `oust` was an empty string (the `src` is the content of an HTML file). I’m sure it wasn’t passed `undefined` because `critical` does a `toString()` on the file contents, and this didn’t fail.
 `type` is also properly passed, as it’s an hardcoded string.
```js
function oust(src, type, raw) {
    if (!src || !type) {
        throw new Error('`src` and `type` required');
    }
```

~It might be that the user has an empty `.html` file. I should probably filter those out.
I might also PR `oust` to do a smarter check than just `!src`, as it becomes `true` when passed an empty file, which it should probably just ignore.~

I PR’d oust: https://github.com/addyosmani/oust/pull/37, I’ll make sure critical is updated as well, so this should be fixed in the next release.

**Comment by ehmicky** [comment]
Thanks for digging into this @Tom-Bonnike!
Strong typing always helps :)

**Comment by Tom-Bonnike** [comment]
I’ll update the plugin as soon as there’s a new version of critical containing the dependency bump and we should hopefully not see this again.

**Comment by Tom-Bonnike** [comment]
Updated critical to 2.0.1 which should fix the issue, released v1.1.2 of the plugin on npm. Going to bump the version on netlify/plugins.
